src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java

```diff
@@ -680,4 +684,56 @@ public ResponseEntity<OrganisationEntityResponse> retrieveOrganisationByUserId(
             @PathVariable("userId") String userId) {
         return organisationService.retrieveOrganisationByUserId(userId);
     }
+
+    @Operation(
+            summary = "Retrieves Organisations by Organisation Profile IDs",
+            description = "**IDAM Roles to access API** : <br> No role restriction",
+            security = {
+                    @SecurityRequirement(name = "ServiceAuthorization")
+            }
+    )
+
+    @ApiResponse(
+            responseCode = "200",
+            description = "List of matching organisations",
+            content = @Content(schema = @Schema(implementation = MultipleOrganisationsResponse.class))
+    )
+    @ApiResponse(
+            responseCode = "400",
+            description = "Invalid request (PageSize or searchAfter) provided",
+            content = @Content(schema = @Schema(implementation = ErrorResponse.class))
+    )
+    @ApiResponse(
+            responseCode = "403",
+            description = "Forbidden Error: Access denied",
+            content = @Content
+    )
+    @ApiResponse(
+            responseCode = "500",
+            description = "Internal Server Error",
+            content = @Content
+    )
+
+    @PostMapping(
+            path = "/getOrganisationsByProfile",
+            produces = APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<Object> retrieveOrganisationsByProfileIds(

```

Comment:
this is from the Jira ticket

```
This is internal only so no IdAM token is required.
On invocation, the endpoint will: 
validate the S2S token, If invalid token or does not correspond to an entry in the allow list.
```

Reply:
OK  - Lets leave it without. But can you put a note in the JIRA description to remind the RD QA to verify this, because its contradictory to what we've been told those /internals are for - thanks